### PR TITLE
Transcripts: handle diacritics when searching

### LINKED ISSUE #2039: Transcript: search should handle diacritics
When searching for `zolc` then `żółć` will be a match. Same for `aviao` and `avião`. Currently in Transcripts this is not a match — and it should be.

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1848 | 
|:---:|

Fixes #2039

Handles diacritics when searching.

I also improved `KMPSearch` to cache the transcript content, avoiding parsing the transcript every time a new search term is entered.

## To test

1. Ensure **transcripts** feature flag is enabled
2. Run the app
3. Play any Chaosradio episode
4. Search for any word with accents or special characters without adding those
5. ✅ The results should match
6. Add a new episode with transcript to your Up Next
7. Move the scrubber to the end of the current episode
8. When the new one starts playing, try searching on it
9. ✅ Search should match correctly

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/57">7.71</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by leandroalonso** [comment]
@danielebogo `ShareDestination.swift` is generating a build error, so you need to add a `return UIImage()` in `static fileprivate func shareImage(_ option: SharingModal.Option, style: ShareImageStyle)`

File: Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift [hunk 1]
```diff
@@ -2,13 +2,12 @@ import Foundation
...
+    private var pattern: [Character] = []
+    private var lps: [Int] = []
+    private var textArray: [String.Element]
 
-    public init(pattern: String) {
-        self.pattern = Array(pattern.lowercased())
-        self.lps = Array(repeating: 0, count: pattern.count)
-        computeLPSArray()
+    public init(text: String) {
+        textArray = Array(text.lowercased().folding(options: .diacriticInsensitive, locale: nil).replacingOccurrences(of: "ł", with: "l"))
```
Comment (danielebogo) [inline Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift]: Nit:
can we wrap `.replacingOccurrences(of: "ł", with: "l")` into an Array extension?

**Review by danielebogo — APPROVED** [review]
All good! Texted it with a German podcast and it worked.
I left just a nit.

File: Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift (hunk 1 above)
Comment (leandroalonso) [reply Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift]: Done in e665abeb9174d03eb47968062d3798c549b5e70d

**Pull request merged**